### PR TITLE
OC-11238 Add API to refresh Randomize configurations without restarti…

### PR DESCRIPTION
**Pull request by yogi-oc**
…ng Runtime

### CODE REVIEW

File: core/src/main/java/org/akaza/openclinica/service/auth/TokenServiceImpl.java [hunk 1]
```diff
@@ -55,4 +56,16 @@
             throw new InvalidTokenException("Cannot convert access token to JSON", e);
         }
     }
+
+    public String getRole(String token) {
```
Comment (svadla-oc) [inline core/src/main/java/org/akaza/openclinica/service/auth/TokenServiceImpl.java]: this should probably be renamed to getUserType since you are returning user type and not role

File: web/src/main/java/org/akaza/openclinica/controller/RandomizeController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    @RequestMapping(value = "/refresh", method = RequestMethod.POST)
+    public ResponseEntity<String> refreshConfiguration(HttpServletRequest request) {
+        String accessToken = utilService.getAccessTokenFromRequest(request);
+        if (StringUtils.isEmpty(accessToken))
+            return new ResponseEntity<>("Access token not specified.", HttpStatus.BAD_REQUEST);
+
+        String role = tokenService.getRole(accessToken);
+
+        if (!StringUtils.equals(role, UserType.TECH_ADMIN.getName()))
+            return new ResponseEntity<>("Tech Admin role is needed to call this service.", HttpStatus.BAD_REQUEST);
```
Comment (svadla-oc) [inline web/src/main/java/org/akaza/openclinica/controller/RandomizeController.java]: we should return a 403 (ACCESS_DENIED) instead of 400

File: web/src/main/java/org/akaza/openclinica/service/randomize/RandomizationServiceImpl.java [hunk 3]
```diff
@@ -83,6 +89,33 @@ public void setConfiguration(RandomizationConfiguration configuration) {
...
+        ArrayList<Study> studies = studyDao.findAll();
+        studies.stream().filter(study->study.getStatus() == Status.AVAILABLE
+                && StringUtils.isNotEmpty(study.getStudyEnvUuid()))
+                .forEach(study-> {
+                    List<ModuleConfigDTO> configs = studyBuildService.getModuleConfigsFromStudyService(accessToken, study);
+                    String moduleEnabled = studyBuildService.isModuleEnabled(configs, study, Modules.RANDOMIZE);
+                    if (StringUtils.equalsIgnoreCase(moduleEnabled, StudyBuildService.ENABLED))
+                        try {
+                            randomizationConfigurations.add(retrieveConfiguration(study.getStudyEnvUuid(), accessToken));
+                        } catch (Exception e) {
```
Comment (svadla-oc) [inline web/src/main/java/org/akaza/openclinica/service/randomize/RandomizationServiceImpl.java]: if there is an error retrieving configuration, should we be returning an error response instead of swallowing the exception?

Comment (yogi-oc) [reply web/src/main/java/org/akaza/openclinica/service/randomize/RandomizationServiceImpl.java]: Logging the exception now.

File: web/src/main/java/org/akaza/openclinica/service/StudyBuildService.java [hunk 4]
```diff
@@ -20,6 +20,10 @@
  */
 public interface StudyBuildService {
     Logger logger = LoggerFactory.getLogger(StudyBuildService.class);
+    String ENABLED = "enabled";
+    String DISABLED = "disabled";
+    String ACTIVE = "active";
+
```
Comment (svadla-oc) [inline web/src/main/java/org/akaza/openclinica/service/StudyBuildService.java]: please move this to ModuleProcessor

File: web/src/main/java/org/akaza/openclinica/service/randomize/RandomizationServiceImpl.java [hunk 5]
```diff
@@ -83,6 +90,41 @@ public void setConfiguration(RandomizationConfiguration configuration) {
...
+        ArrayList<Study> studies = studyDao.findAll();
+        studies.stream().filter(study->study.getStatus() == Status.AVAILABLE
+                && StringUtils.isNotEmpty(study.getStudyEnvUuid()))
+                .forEach(study-> {
+                    List<ModuleConfigDTO> configs = studyBuildService.getModuleConfigsFromStudyService(accessToken, study);
+                    String moduleEnabled = studyBuildService.isModuleEnabled(configs, study, Modules.RANDOMIZE);
+                    if (StringUtils.equalsIgnoreCase(moduleEnabled, StudyBuildService.ENABLED))
+                        try {
+                            randomizationConfigurations.add(retrieveConfiguration(study.getStudyEnvUuid(), accessToken));
+                            results.put(study.getOc_oid(), SUCCESS_STR);
```
Comment (svadla-oc) [inline web/src/main/java/org/akaza/openclinica/service/randomize/RandomizationServiceImpl.java]: please move this out of the try block

File: core/src/main/resources/org/akaza/openclinica/applicationContext-core-hibernate.xml [hunk 6]
```diff
@@ -137,7 +137,7 @@
 
         <property name="hibernateProperties">
             <props>
-                <prop key="hibernate.show_sql">false</prop>
+                <prop key="hibernate.show_sql">true</prop>
```
Comment (svadla-oc) [inline core/src/main/resources/org/akaza/openclinica/applicationContext-core-hibernate.xml]: revert this please

File: core/src/main/java/org/akaza/openclinica/service/JobTriggerService.java [hunk 7]
```diff
@@ -74,8 +74,8 @@ public JobTriggerService(DataSource ds, RuleSetDao ruleSetDao, RuleSetService ru
 	}
 
 	// @Scheduled(cron = "0 0/2 * * * ?") // trigger every 2 minutes
-	// @Scheduled(cron = "0 0/1 * * * ?") // trigger every minute
-	 @Scheduled(cron = "0 0 0/1 * * ?") // trigger every hour
+	 @Scheduled(cron = "0 0/1 * * * ?") // trigger every minute
+	// @Scheduled(cron = "0 0 0/1 * * ?") // trigger every hour
```
Comment (svadla-oc) [inline core/src/main/java/org/akaza/openclinica/service/JobTriggerService.java]: what's the reason for this change?

**Comment by svadla-oc** [comment]
👍
